### ISSUE TITLE
Allow torrents to be added through POST

### ISSUE THREAD

**Issue by jvacek** [jfurrow/flood#580]
This feature was available using rutorrent, a specific private tracker, a user script and tampermonkey.
It basically adds an extra link next to the torrent-download link that sends the torrent straight to rutorrent.
The userscript itself is located here and actual useful code seems to start at line 196 https://pastebin.com/2TePhv2u. Runs very smoothly on pretty much all Gazelle-based private trackers.
The generated link then looks something like the following:

https://_example.com_/rutorrent/php/addtorrent.php?url=https%3A//_tracker.org_/torrents.php%3Faction%3Ddownload%26id%_idnumber_%26authkey%_3Daspecificstring_%26torrent_pass%3D_anotherstring_

Would be sweet if a similar "API" could be developed in order to send torrents straight to flood using user-scripts.

**Comment by dcousens** [comment]
Authentication required

**Comment by jvacek** [comment]
@dcousens indeed, rutorrent allows for this as you can auth using nginx/apache auth. Would it be possible to somehow use the cookie for this?

**Comment by CluelessTechnologist** [comment]
Is this what the FireFox addon Torrent to web uses?
https://github.com/DASPRiD/Torrent-to-Web

How else is it able to send a torrent file directly to ruTorrent?

**Comment by jvacek** [comment]
@CluelessTechnologist yes https://github.com/DASPRiD/Torrent-to-Web/blob/master/src/background/adapter/rutorrent.js

I guess a possibility would also be an API. If you could auth on that level, make a JSON and send it over, done deal.

**Comment by jesec** [comment]
Implemented by d97a2c5a9519c1ae480cf121ab8311c519262674. 

You may close the issue if it is no longer relevant.
